### PR TITLE
RESTWS-707: Fix bug in encounterRole endpoint

### PR DESCRIPTION
**Pull request by larrystone**
Fix bug with `encounterRole` endpoint causing empty set `[ ]` to be returned when a query paramter in provider in API call
e.g `v1/encounterRole?q=unknown`

Issue: [RESTWS-707](https://issues.openmrs.org/browse/RESTWS-707)

### CODE REVIEW

**Comment by dkayiwa** [comment]
@larrystone did you take a look at the ticket description? It has the exact classes and changes that you need.

File: omod-1.11/src/test/java/org/openmrs/module/webservices/rest/web/RestTestConstants1_11.java [hunk 1]
```diff
@@ -12,4 +12,6 @@
 public class RestTestConstants1_11 {
 	
 	public static final String DRUG_UUID = "05ec820a-d297-44e3-be6e-698531d9dd3f";
+
```
Comment (dkayiwa) [inline omod-1.11/src/test/java/org/openmrs/module/webservices/rest/web/RestTestConstants1_11.java]: Is this used anywhere?

File: omod-1.11/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_11/EncounterRoleController1_11.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import org.openmrs.EncounterRole;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.test.Util;
+import org.openmrs.module.webservices.rest.web.RestTestConstants1_9;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+public class EncounterRoleController1_11 extends MainResourceControllerTest {
```
Comment (dkayiwa) [inline omod-1.11/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_11/EncounterRoleController1_11.java]: Can you update this class name to be like the one i put in the ticket description?

File: omod-1.11/src/test/java/org/openmrs/module/webservices/rest/web/RestTestConstants1_11.java (hunk 1 above)
Comment (larrystone) [reply omod-1.11/src/test/java/org/openmrs/module/webservices/rest/web/RestTestConstants1_11.java]: Yes, it's been used in the test files

Comment (dkayiwa) [reply omod-1.11/src/test/java/org/openmrs/module/webservices/rest/web/RestTestConstants1_11.java]: Can you point me to the exact line?

Comment (larrystone) [reply omod-1.11/src/test/java/org/openmrs/module/webservices/rest/web/RestTestConstants1_11.java]: https://github.com/larrystone/openmrs-module-webservices.rest/blob/41b70a6fe09f2601ed7be96c2204ea8a73bea097/omod-1.11/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_11/EncounterRoleResource1_11Test.java#L52

https://github.com/larrystone/openmrs-module-webservices.rest/blob/41b70a6fe09f2601ed7be96c2204ea8a73bea097/omod-1.11/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_11/EncounterRoleController1_11Test.java#L33

File: omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/RestTestConstants1_9.java [hunk 3]
```diff
@@ -51,8 +51,6 @@
 	
 	public static final String CONCEPT_UUID = "c607c80f-1ea9-4da3-bb88-6276ce8868dd";
 	
-	public final static String ENCOUNTER_ROLE_UUID = "a0b03050-c99b-11e0-9572-0800200c9a66";
```
Comment (dkayiwa) [inline omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/RestTestConstants1_9.java]: Why did you remove this?

Comment (larrystone) [reply omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/RestTestConstants1_9.java]: It was being duplicated, I already have it in the `RestTestConstans1_11.java` file

**Comment by dkayiwa** [comment]
Did you take a look at the travis failure?

**Comment by larrystone** [comment]
I am looking into it.I have a feeling it's because I am on a slow network. I will try using a LAN network tomorrow. I also have some files that were autogenerated when I ran `mvn clean install`. is it okay to add them to my git files?
screen shot 2018-05-10 at 8 58 58 pm

**Comment by dkayiwa** [comment]
Just leave the auto modified files.
As for the travis failure, the log should give you a clue in regards to what you have not done correctly.

**Comment by larrystone** [comment]
I will look into it. Thanks @dkayiwa

**Comment by larrystone** [comment]
screen shot 2018-05-10 at 10 39 38 pm
This is the error I am getting now yet all tests are passing locally

**Comment by larrystone** [comment]
@dkayiwa The tests are now passing but I noticed coverage decreased even though the file I created had 100% coverage. Please is there anything I can do about this?
screen shot 2018-05-11 at 1 41 13 am
